### PR TITLE
Output warning on filter error

### PR DESCRIPTION
**Pull request by springmeyer**
See #45. This address the problem by outputting a warning like:

> [vtshaver] filter error: 'Expected value to be of type number, but found null instead.' skipping feature 327765642 in poi_label

Because I'm not confidence throwing is the right solution here I think a warning is an improvement and safe.

### CODE REVIEW

File: src/shave.cpp [hunk 1]
```diff
@@ -354,9 +358,22 @@ void filterFeatures(vtzero::tile_builder* finalvt,
...
+        // expression hit error
+        if (!result) {
+            std::ostringstream msg;
+            msg << "[vtshaver] filter error: '" << result.error().message << "' skipping feature";
+            if (feature.has_id()) {
+              msg << " " << feature.id();
+            }
+            msg << " in " << std::string{layer.name()};
+            std::clog << msg.str() << "\n";
+            return true;
```
Comment (springmeyer) [inline src/shave.cpp:371]: This `return true` means that the feature will still be dropped (maintaining current behavior).

This new log will very useful for folks that use the `vtshaver` on the command line.

When vtshaver is deployed in an API-like environment this warning will not likely be useful, so I should likely add a way to turn it off as a future followup.

File: src/shave.cpp [hunk 2]
```diff
@@ -354,9 +358,22 @@ void filterFeatures(vtzero::tile_builder* finalvt,
...
+        }
+
+        const mbgl::optional<bool> typed = mbgl::style::expression::fromExpressionValue<bool>(*result);
+        bool match = typed ? *typed : false;
```
Comment (springmeyer) [inline src/shave.cpp:375]: This fancy type logic is taken/matches https://github.com/mapbox/mapbox-gl-native/blob/647959a85327660a8d0b35da0b8a247820a36c15/src/mbgl/style/filter.cpp#L14.

**Comment by mapsam** [comment]
@springmeyer any idea what the failing test is about?

**Comment by springmeyer** [comment]
> @springmeyer any idea what the failing test is about?

Yes, it's a bus error on the CFI job. I can replicate locally using CFI https://clang.llvm.org/docs/ControlFlowIntegrity.html. Either it is some edge case bug in my implementation or CFI spurious error. 

I'll narrow it down and fix it - one way or another - before this merges.
